### ISSUE TITLE
Rule Change: Update `no-prototype-builtins` to Allow Builtins on Native JS Classes

### ISSUE THREAD

**Issue by ITenthusiasm** [eslint/eslint#17203]
### What rule do you want to change?

`no-prototype-builtins`

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
// app/helpers/safelySetHTML.js

let sanitizeHTML;
if(!Element.prototype.hasOwnProperty("setHTML")) {
  import("dompurify").then(({ sanitize }) => (sanitizeHTML = sanitize));
}

export default function safelySetHTML(element, markup) {
  element.setHTML?.(markup) || sanitizeHTML(markup);
}
```

### What does the rule currently do for this code?

I receive the following error while checking `Element.prototype.hasOwnProperty("setHTML")`:

> Do not access Object.prototype method 'hasOwnProperty' from target object. eslint([`no-prototype-builtins`](https://eslint.org/docs/rules/no-prototype-builtins))

### What will the rule do after it's changed?

The rule will allow native JavaScript classes to leverage the methods restricted by `no-prototype-builtins`, such as `hasOwnProperty`. Basically, the code sample above (and similar pieces of code) would be legal. Otherwise, the rule would generate a warning as usual.

If I understand correctly, this should be safe for all JS classes. If that's is true, then the current `no-prototype-builtins` rule may be too overreaching, and a change like this would fix that issue.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

I've never submitted a PR to ESLint before, so as a heads up I'd probably need a couple quick pointers in the right direction. (I imagine some docs may already exist?)

**Comment by Rec0iL99** [comment]
Hi @ITenthusiasm, thanks for the issue. According to https://github.com/eslint/eslint/issues/2693#issuecomment-109071075  in the original proposal, I don't think it's possible to track how the object was created (whether it used `Object.create` or not). The @eslint/eslint-tsc folks will have better insight into this.

**Comment by nzakas** [comment]
> Hi @ITenthusiasm, thanks for the issue. According to https://github.com/eslint/eslint/issues/2693#issuecomment-109071075 in the original proposal for no-prototype-builtins, I don't think it's possible to track how the object was created (whether it was created by Object.create or not). The https://github.com/orgs/eslint/teams/eslint-tsc folks will have better insight into this.

Yes, that's correct. ESLint doesn't have the ability to track objects through their lifecycle, so we don't know how it was created, modified, etc., in order to determine how to treat it. To do so correctly would be a lot of work (including going cross-file in the case of imports), so we aren't exploring that.

**Comment by ITenthusiasm** [comment]
@nzakas Understood. Thank you!
